schema-registry-mock/src/main/java/com/bakdata/schemaregistrymock/SchemaRegistryMock.java

```diff
@@ -18,74 +43,94 @@
+            WireMockConfiguration.wireMockConfig().dynamicPort().extensions(this.autoRegistrationHandler));
     private final SchemaRegistryClient schemaRegistryClient = new MockSchemaRegistryClient();
 
     @Override
-    public void afterEach(ExtensionContext context) {
-        this.httpMock.stop();
+    public void afterEach(final ExtensionContext context) {
+        this.mockSchemaRegistry.stop();
     }
 
     @Override
-    public void beforeEach(ExtensionContext context) {
-        this.httpMock.start();
-        this.httpMock.stubFor(WireMock.post(WireMock.urlPathMatching("/subjects/[^/]+/versions"))
-                .willReturn(WireMock.aResponse().withTransformers(this.registrationHandler.getName())));
-        this.httpMock.stubFor(WireMock.get(WireMock.urlPathMatching("/schemas/ids/.*")).willReturn(WireMock.aResponse().withStatus(HTTP_NOT_FOUND)));
+    public void beforeEach(final ExtensionContext context) {
+        this.mockSchemaRegistry.start();
+        this.mockSchemaRegistry.stubFor(WireMock.post(WireMock.urlPathMatching(SCHEMA_REGISTRATION_PATTERN))
+                .willReturn(WireMock.aResponse().withTransformers(this.autoRegistrationHandler.getName())));
+        this.mockSchemaRegistry.stubFor(WireMock.get(WireMock.urlPathMatching(SCHEMA_BY_ID_PATTERN + "*"))
+                .willReturn(WireMock.aResponse().withStatus(HTTP_NOT_FOUND)));
     }
 
-    public int register(String subject, Schema schema) throws IOException, RestClientException {
-        final int id = this.schemaRegistryClient.register(subject, schema);
-        this.httpMock.stubFor(WireMock.get(WireMock.urlEqualTo("/schemas/ids/" + id)).willReturn(ResponseDefinitionBuilder.okForJson(new SchemaString(schema.toString()))));
-        log.debug("Registered schema " + id);
-        return id;
+    public int registerKeySchema(final String topic, final Schema schema) {
+        return this.register(topic + "-key", schema);
+    }
+
+    public int registerValueSchema(final String topic, final Schema schema) {
+        return this.register(topic + "-value", schema);
+    }
+
+    private int register(final String subject, final Schema schema) {
+        try {
+            final int id = this.schemaRegistryClient.register(subject, schema);
+            this.mockSchemaRegistry.stubFor(WireMock.get(WireMock.urlEqualTo(SCHEMA_BY_ID_PATTERN + id))
+                    .willReturn(ResponseDefinitionBuilder.okForJson(new SchemaString(schema.toString()))));
+            log.debug("Registered schema {}", id);
+            return id;
+        } catch (final IOException | RestClientException e) {
+            throw new IllegalStateException("Internal error in mock schema registry client", e);
+        }
     }
 
     public SchemaRegistryClient getSchemaRegistryClient() {
-        return new CachedSchemaRegistryClient(url(), 1000);
+        return new CachedSchemaRegistryClient(this.getUrl(), 1000);
```

Comment:
i assume this is a cache size? nonetheless, either comment or make it a variable.